src/main/java/org/myrobotlab/service/Servo.java

```diff
@@ -88,6 +86,18 @@ public Servo(String n, String id) {
   @Override
   protected boolean processMove(Double newPos, boolean blocking, Long timeoutMs) {
 
+    if (newPos == null) {
+      log.info("servo processMove(null) not valid position");
+      return false;
+    }
+
+    double minLimit = Math.min(mapper.minX, mapper.maxX);
+    double maxLimit = Math.max(mapper.minX, mapper.maxX);
+    newPos = (newPos < minLimit) ? minLimit : newPos;

```

Comment:
Can be replaced with Math.max()

Or, we add a forceToRange() method in some utility class to take care of this and the next line

Reply:
maybe later :thinking: